### PR TITLE
feat(internal/gensnippets): initial commit

### PR DESCRIPTION
**Pull request by tbpg**
This tool renders all examples in the current directory to standalone
files.

I confirmed this successfully processes every example in this repo. Once
I added a go.mod/go.sum in the generated directory, all of the examples
passed go vet.

A future change will integrate this tool into our automatic generation
pipelines and add region tags.

### CODE REVIEW

File: internal/gensnippets/main.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		if err != nil {
+			return err
+		}
+		if d.Name() == "go.mod" {
+			dirs = append(dirs, filepath.Dir(path))
+		}
+		return nil
+	})
+
+	fmt.Printf("Processing %v modules\n", len(dirs))
```
Comment (codyoss) [inline internal/gensnippets/main.go:55]: nit: log?

File: internal/gensnippets/main.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+		if strings.HasPrefix(s, "{\n") && strings.HasSuffix(s, "\n}") {
+			lines := strings.Split(s, "\n")
+			builder := strings.Builder{}
+			for _, line := range lines[1 : len(lines)-1] {
+				builder.WriteString(strings.TrimPrefix(line, "\t"))
+				builder.WriteString("\n")
+			}
+			s = builder.String()
+		}
+		if err := os.MkdirAll(dir, 0755); err != nil {
```
Comment (codyoss) [inline internal/gensnippets/main.go:140]: Does this need the execute bit?

**Review by codyoss — APPROVED** [review]
Overall LGTM, just a couple small things.

**Review by tbpg — COMMENTED** [review]
Thanks!

**Review by noahdietz — APPROVED** [review]
lgtm

**Pull request merged**